src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -1232,15 +1232,29 @@ else if (MISSING_CHANNEL.equals(ex.getMessage()))
 
         if (type == MessageType.UNKNOWN)
             throw new IllegalArgumentException(UNKNOWN_MESSAGE_TYPE);
+
+        MessageReference messageReference = null;
+
+        if (!jsonObject.isNull("message_reference"))
+        {
+            DataObject messageReferenceJson = jsonObject.getObject("message_reference");
+
+            messageReference = new MessageReference(
+                    messageReferenceJson.getLong("message_id"),
+                    messageReferenceJson.getLong("channel_id"),
+                    referencedMessage
+            );
+        }
```

Comment:
Lets move this up closer to the `referenced_message` code
Lets also add some inline documentation explaining (briefly) the difference between `referenced_message` and `message_reference`